gxa/src/test/java/uk/ac/ebi/atlas/controllers/rest/DifferentialJsonResultsParserTest.java

```diff
@@ -0,0 +1,186 @@
+package uk.ac.ebi.atlas.controllers.rest;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import uk.ac.ebi.atlas.search.diffanalytics.DiffAnalytics;
+
+import uk.ac.ebi.atlas.trader.ContrastTrader;
+
+import java.util.List;
+
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.assertThat;
+
+@RunWith(MockitoJUnitRunner.class)
+public class DifferentialJsonResultsParserTest {
+
+    private DifferentialJsonResultsParser subject;
+
+    @Rule
+    public ExpectedException thrown= ExpectedException.none();
+
+    @Mock
+    private ContrastTrader contrastTraderMock;
+
+    @Before
+    public void setUp() throws Exception {
+
+        subject = new DifferentialJsonResultsParser(contrastTraderMock);
+    }
+
+    @Test
+    public void invalidJsonResultThrows() throws Exception {
+
+        thrown.expect(RuntimeException.class);
+        thrown.expectMessage("Input Json is missing a results field OR results field is not an Array");
+
+        JsonObject result = new JsonObject();
+        result.addProperty("results",1);
+
+        List<DiffAnalytics> expression = subject.parseDifferentialResults(result);
+    }
+
+    @Test
+    public void emptyJsonResultArrayProducesEmptyList() throws Exception {
+
+        JsonObject result = new JsonObject();
+
+        result.add("results", new JsonArray());
+
+        List<DiffAnalytics> expression = subject.parseDifferentialResults(result);
+        assertThat(expression,hasSize(0));
+    }
+
+    @Test
+    public void numberOfJsonObjectsInResultArrayProducesEqualLengthList() throws Exception {
+
+        JsonObject result = new JsonObject();
+        JsonArray results = new JsonArray();
+
+        for(int i = 1; i<=3; i++) {
+
+            JsonObject element = new JsonObject();
+            element.addProperty("experiment_accession", "e"+ i);
+            element.addProperty("contrast_id", "c"+ i);
+            element.addProperty("bioentity_identifier", "id"+ i);
+            element.addProperty("species", "org"+ i);
+            element.addProperty("p_value", 1.11);
+            element.addProperty("fold_change", -1.12);
+            element.addProperty("t_statistic", 0.0);
+            element.addProperty("keyword_symbol", i);
+            results.add(element);
+        }
```

Comment:
Refactor this loop into a utilitiy method that returns synthetic test data of any given size instead.